### PR TITLE
Remove option to load new blocks from ConnectTip

### PR DESCRIPTION
**Pull request by nuttycom**
A minor refactoring to move file IO out of ConnectTip, which has only a single caller.

### CODE REVIEW

**Comment by str4d** [comment]
@zkbot try

**Comment by ebfull** [comment]
ACK

File: src/main.cpp [hunk 1]
```diff
@@ -3122,15 +3122,8 @@ uint64_t nNotifiedSequence = 0;
  */
 bool static ConnectTip(CValidationState& state, const CChainParams& chainparams, CBlockIndex* pindexNew, const CBlock* pblock)
 {
-    assert(pindexNew->pprev == chainActive.Tip());
-    // Read block from disk.
+    assert(pblock && pindexNew->pprev == chainActive.Tip());
```
Comment (str4d) [inline src/main.cpp:3125]: This assertion is being triggered in the RPC tests. Needs investigation.

**Comment by nuttycom** [comment]
@zkbot try

File: src/main.cpp [hunk 2]
```diff
@@ -3341,7 +3334,18 @@ static bool ActivateBestChainStep(CValidationState& state, const CChainParams& c
 
         // Connect new blocks.
         BOOST_REVERSE_FOREACH(CBlockIndex *pindexConnect, vpindexToConnect) {
-            if (!ConnectTip(state, chainparams, pindexConnect, pindexConnect == pindexMostWork ? pblock : NULL)) {
+            const CBlock* pconnectBlock;
+            CBlock block;
+            if (pblock && pindexConnect == pindexMostWork) {
```
Comment (nuttycom) [inline src/main.cpp:3339]: @str4d Adding a check that `pblock != NULL` here is necessary, I think, but perhaps not sufficient; it's probably important to check that pblock is also actually related to pindexMostWork - we have a comment at the start of the procedure that says that should be respected, but then nothing that actually enforces that invariant?

**Comment by nuttycom** [comment]
@zkbot try

**Comment by ebfull** [comment]
@zkbot try

File: src/main.cpp (hunk 2 above)
Comment (str4d) [reply src/main.cpp:3429]: Enforcing this invariant would be nice, but that's a much larger refactor. This is a good separate step.

**Comment by daira** [comment]
@zkbot r+

**Pull request merged**